bindings/python/src/encoding.rs

```diff
@@ -73,6 +54,18 @@ impl PyEncoding {
         }
     }
 
+    fn __repr__(&self) -> PyResult<String> {
```

Comment:
`#[pyproto]` is deprecated in pyo3 0.16: https://pyo3.rs/main/migration.html#pyproto-has-been-deprecated , `#[pymethods]` is the replacement.